app/abilities/api_capability.rb

```diff
@@ -5,11 +5,7 @@ class ApiCapability
 
   def initialize(token)
     if Settings.status != "database_offline"
-      user = if token.respond_to?(:resource_owner_id)
-               User.find(token.resource_owner_id)
-             elsif token.respond_to?(:user)
-               token.user
-             end
+      user = (User.find(token.resource_owner_id) if token.respond_to?(:resource_owner_id))
```

Comment:
We don't actually need the condition at all - we should always have a token and it should always be an OAuth 2 token now.

There is a test that will need fixing but it's a bogus test that deliberately passes `nil` as the token which is not valid. The real code never does that.

Reply:
Removed that part of the test because you don't have any abilities by default and it only tested `cannot?`.